proposals/NativeInt.md

```diff
@@ -0,0 +1,247 @@
+# Native-sized integers
+
+## Summary
+[summary]: #summary
+
+Language support for a native-sized signed and unsigned integer types.
+
+The motivation is for interop scenarios and for low-level libraries.
+
+## Design
+[design]: #design
+
+New contextual keywords `nint` and `nuint` represent native signed and unsigned integer types.
+The identifiers are only treated as keywords when used in a type context and when the identifier does not otherwise bind to a symbol at that program location.
+
+The types `nint` and `nuint` are essentially aliases to underlying types `System.IntPtr` and `System.UIntPtr`, where the compiler can surface additional conversions and operations for native ints.
+
+### Constants
+
+There is no direct syntax for native int literals. Explicit casts of other integral constant values can be used instead: `(nint)42`.
```

Comment:
Given that there is an implicit conversion from `int` to `nint`, `nint x = 42` and `nint x = (nint)42` would both be valid, correct?

I presume `const nint x = 42` would also work?

Reply:
Yes, `const nint x = 42` would work.

---
In reply to: [329172165](https://github.com/dotnet/csharplang/pull/2833#discussion_r329172165) [](ancestors = 329172165)